storage/key_index.go

```diff
@@ -221,3 +240,20 @@ func (g *generation) walk(f func(rev reversion) bool) int {
 func (g *generation) String() string {
 	return fmt.Sprintf("g: ver[%d], revs %#v\n", g.ver, g.revs)
 }
+
+func (a generation) equal(b generation) bool {
+	if a.ver != b.ver {
+		return false
```

Comment:
reversion is the global reversion of the entire versioned storage.

version is the version of the key. 

For example, we have a key foo.

set foo -> bar  global rev:1; key foo version: 1

set foo->bar 1 global rev:2, key foo version: 2

rm foo global rev:3, key foo version: null

set foo->bar global rev:4; key foo version: 1